dask/dataframe/multi.py

```diff
@@ -1268,12 +1272,14 @@ def concat(
         raise ValueError("'join' must be 'inner' or 'outer'")
 
     axis = DataFrame._validate_axis(axis)
-    try:
-        # remove any empty DataFrames
-        dfs = [df for df in dfs if bool(len(df.columns))]
-    except AttributeError:
-        # 'Series' object has no attribute 'columns'
-        pass
+
+    if axis == 1:
+        try:
+            # remove any empty DataFrames
+            dfs = [df for df in dfs if bool(len(df.columns))]
```

Comment:
In PR description, you mention that:

> we should keep DataFrames with no columns but an Index"

This code doesn't seem to be doing it, and I don't see a test case for it. Can you clarify?

Reply:
I forgot one important part:

> we should keep DataFrames with no columns but an Index for axis=0

Reason is simple: Consistency with pandas